### PR TITLE
🐛 Fix cluster topology and UpdateCoreDNS test flakes

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

Co-authored-by: killianmuldoon <kmuldoon@vmware.com>
Co-authored-by: Yuvaraj Kakaraparthi <kakaraparthy@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: controllers/topology/cluster_controller_test.go [hunk 1]
```diff
@@ -547,7 +547,7 @@ func setupTestEnvForIntegrationTests(ns *corev1.Namespace) (func() error, error)
 	cleanup := func() error {
 		// Delete Objects in reverse, because we cannot delete a ClusterCLass if it is still used by a Cluster.
 		for i := len(initObjs) - 1; i >= 0; i-- {
-			if err := env.Delete(ctx, initObjs[i]); err != nil {
+			if err := env.CleanupAndWait(ctx, initObjs[i]); err != nil {
```
Comment (sbueringer) [inline controllers/topology/cluster_controller_test.go:550]: Sometimes the Cluster was not gone in the local cache and then the subsequent ClusterClass deletion failed

File: controlplane/kubeadm/internal/workload_cluster_coredns_test.go [hunk 2]
```diff
@@ -422,10 +422,13 @@ func TestUpdateCoreDNS(t *testing.T) {
...
-				g.Expect(env.Get(ctx, client.ObjectKey{Name: kubeadmConfigKey, Namespace: metav1.NamespaceSystem}, &expectedKubeadmConfigMap)).To(Succeed())
-				g.Expect(expectedKubeadmConfigMap.Data).To(HaveKeyWithValue("ClusterConfiguration", ContainSubstring(tt.kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.DNS.ImageTag)))
-				g.Expect(expectedKubeadmConfigMap.Data).To(HaveKeyWithValue("ClusterConfiguration", ContainSubstring(tt.kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.DNS.ImageRepository)))
+				g.Eventually(func(g Gomega) error {
+					var expectedKubeadmConfigMap corev1.ConfigMap
+					g.Expect(env.Get(ctx, client.ObjectKey{Name: kubeadmConfigKey, Namespace: metav1.NamespaceSystem}, &expectedKubeadmConfigMap)).To(Succeed())
+					g.Expect(expectedKubeadmConfigMap.Data).To(HaveKeyWithValue("ClusterConfiguration", ContainSubstring(tt.kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.DNS.ImageTag)))
+					g.Expect(expectedKubeadmConfigMap.Data).To(HaveKeyWithValue("ClusterConfiguration", ContainSubstring(tt.kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.DNS.ImageRepository)))
+					return nil
+				}, "5s").Should(Succeed())
```
Comment (sbueringer) [inline controlplane/kubeadm/internal/workload_cluster_coredns_test.go:431]: Sometimes the local cache was stale, so it still checked the old version of the COnfigMap

**Comment by sbueringer** [comment]
/cc @killianmuldoon @ykakarap 
Thx for debugging / fixing this together!

/assign @fabriziopandini

**Review by killianmuldoon — COMMENTED** [review]
😄 

/lgtm

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**